.github/workflows/ci.yaml

```diff
@@ -79,47 +84,47 @@ jobs:
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
-      - name: Set up micromamba
-        uses: mamba-org/setup-micromamba@f8b8a1e23a26f60a44c853292711bacfd3eac822
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
         with:
-          environment-file: environment.yml
-          cache-downloads: true
-          cache-environment: true
-          cache-environment-key: ${{ runner.os }}-${{ hashFiles('environment.yml') }}-${{ matrix.PYTHON_VERSION }}-${{ matrix.SA_VERSION }}
-          create-args: >-
-            python=${{ matrix.PYTHON_VERSION }}
-            sqlalchemy${{ matrix.SA_VERSION }}
-            pyodbc
+          environments: ${{ matrix.env }}
       # TODO: Consider removing this manual installation since the docker image already comes with a msodbc driver.
       - name: Install msodbcsql17 driver
-        shell: bash -l {0}
         run: |
           wget https://packages.microsoft.com/ubuntu/20.04/prod/pool/main/m/msodbcsql17/msodbcsql17_17.9.1.1-1_amd64.deb
           ACCEPT_EULA=Y sudo apt install ./msodbcsql17_17.9.1.1-1_amd64.deb --allow-downgrades
       - name: Wait for SQL Server
         timeout-minutes: 1
         run: until docker logs "${{ job.services.db.id }}" 2>&1 | grep -q "SQL Server is now ready"; do sleep 10; done
-      - name: Run tests
-        uses: ./.github/actions/pytest
+      - run: |
+          pixi run -e ${{ matrix.env }} postinstall
+          pixi run -e ${{ matrix.env }} coverage --color=yes
+      - name: Generate code coverage report
+        uses: codecov/codecov-action@v3.1.3
         with:
-          backend: mssql
-          args: tests/integration
+          file: ./coverage.xml
 
   postgres:
     if: ${{ contains(github.event.pull_request.labels.*.name, 'postgres') || contains(github.event.pull_request.labels.*.name, 'ready') || github.ref == 'refs/heads/main' }}
     name: "PostgreSQL ${{ matrix.POSTGRES_VERSION }} - Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }}"
     runs-on: ubuntu-20.04
-    env:
-      CI: True
     strategy:
       fail-fast: false
       matrix:
-        PYTHON_VERSION: [ '3.8', '3.9', '3.10' ]
-        POSTGRES_VERSION: [ '11', 'latest' ]
-        SA_VERSION: ["<2.0", ">=2.0"]
+        postgres_version:
+          - latest
+          - 11
+        env:
+          - postgres-py38
```

Comment:
This doesn't look complete. We should have the cross product of python, postgres and sqlalchemy versions. Here it looks like python and sqlalchemy are decoupled now. 

This observation applies to the others too.

Reply:
i think testing against the border of the cross product matrix should be fine as well, otherwise this can easily explode. i also don't think we would catch significantly more errors that way 🤔

Reply:
We're aware it can explode. That's why in the obscure databases we covered a few combinations.

We can discuss how we effectively do integration tests. Doing the borders of the testing phase space is fine by me.